src/main/java/org/craftercms/studio/impl/v1/service/webdav/WebDavServiceImpl.java

```diff
@@ -180,22 +180,32 @@ public String upload(@ValidateStringParam(name = "site_id") final String site,
                          final InputStream content)
         throws WebDavException {
         WebDavProfile profile = profileReader.getProfile(site, profileId);
-        String uploadUrl = profile.getBaseUrl();
+        String uploadUrl = StringUtils.removeEnd(profile.getBaseUrl(), "/");
         try {
+            Sardine sardine = SardineFactory.begin(profile.getUsername(), profile.getPassword());
+
             if(StringUtils.isNotEmpty(path)) {
-                uploadUrl +=  path.startsWith("/")? path : "/" + path;
+                String[] folders = StringUtils.split(path, "/");
+
+                for(String folder : folders) {
+                    uploadUrl += "/" + folder;
+
+                    try {
+                        logger.debug("Trying to create folder {0}", uploadUrl);
+                        sardine.createDirectory(uploadUrl);
+                        logger.debug("Folder created");
+                    } catch (Exception e) {
+                        logger.debug("Folder already exists");
```

Comment:
(Same as with PR 1339) Wouldn't it be better to first check if the dir exists and then create it? Instead of trying to create it and then ignore the exception if it already exists. Also what if the exception is a connection issue?